AK/Variant.h

```diff
@@ -40,47 +39,134 @@ consteval IndexType index_of()
     return VariantIndexOf<T, IndexType, 0, Ts...> {}();
 }
 
+template<typename... T>
+union VariantStorage;
+
+template<typename T, typename... Rest>
+union VariantStorage<T, Rest...> {
+    VariantStorage() = default;
+    VariantStorage()
+    requires(!IsTriviallyConstructible<T> || (!IsTriviallyConstructible<Rest> || ...)) {};
+
+    VariantStorage(VariantStorage&&) = default;
+    VariantStorage(VariantStorage const&) = default;
+
+    VariantStorage& operator=(VariantStorage&&) = default;
+    VariantStorage& operator=(VariantStorage const&) = default;
```

Comment:
Won't those be invalid for sufficiently non-trivial types?

Reply:
Those should only get initialized -and there for checked- iff the default comparison operator on Variant proper is used, the non trivial case uses a `visit` and `get_pointer()`

Reply:
Ah, sorry I can't read, thought this was `operator==`, will check again